bindings/netframework/ElectionGuard.NetFramework/ElectionGuardInterop/Ballot.h

```diff
@@ -1,4 +1,413 @@
+    ref class PlaintextBallotSelection : ManagedInstance<electionguard::PlaintextBallotSelection>
+    {
+      public:
+        PlaintextBallotSelection(String ^ objectId, uint64_t vote)
+            : PlaintextBallotSelection(objectId, vote, false, nullptr)
+        {
+        }
+        PlaintextBallotSelection(String ^ objectId, uint64_t vote, bool isPlaceholder)
+            : PlaintextBallotSelection(objectId, vote, isPlaceholder, nullptr)
+        {
+        }
+        PlaintextBallotSelection(String ^ objectId, uint64_t vote, bool isPlaceholder,
+                                 ExtendedData ^ extendedData)
+        {
+            std::string _objectId;
+            Utilities::MarshalString(objectId, _objectId);
+
+            if (extendedData != nullptr) {
+                this->_instance = new electionguard::PlaintextBallotSelection(
+                  _objectId, vote, isPlaceholder,
+                  std::make_unique<electionguard::ExtendedData>(*extendedData->_instance));
+                return;
+            }
+            this->_instance =
+              new electionguard::PlaintextBallotSelection(_objectId, vote, isPlaceholder);
+        }
+        internal : PlaintextBallotSelection(
+                     std::unique_ptr<electionguard::PlaintextBallotSelection> other)
+        {
+            this->_instance = other.release();
+        }
+
+        PlaintextBallotSelection(electionguard::PlaintextBallotSelection *unowned)
+            : ManagedInstance(unowned, false)
+        {
+        }
+
+      public:
+        property String ^
+          ObjectId {
+              String ^ get() {
+                  auto value = this->_instance->getObjectId();
+                  return gcnew String(value.c_str());
+              }
+          }
+
+          property uint64_t ^
+          Vote { uint64_t ^ get() { return this->_instance->getVote(); } }
+
+          property bool ^
+          IsPlaceholder { bool ^ get() { return this->_instance->getIsPlaceholder(); } }
+
+          bool IsValid(String ^ expectedObjectId)
+        {
+            std::string _expectedObjectId;
+            Utilities::MarshalString(expectedObjectId, _expectedObjectId);
+            return this->_instance->isValid(_expectedObjectId);
+        }
+    };
+
+  public
+    ref class CiphertextBallotSelection : ManagedInstance<electionguard::CiphertextBallotSelection>
+    {
+      public:
+        CiphertextBallotSelection(String ^ objectId, ElementModQ ^ descriptionHash,
+                                  ElGamalCiphertext ^ ciphertext, bool isPlaceholder,
+                                  ElementModQ ^ nonce, ElementModQ ^ cryptoHash,
+                                  DisjunctiveChaumPedersenProof ^ proof,
+                                  ElGamalCiphertext ^ extendedData)
+        {
+            std::string _objectId;
+            Utilities::MarshalString(objectId, _objectId);
+
+            this->_instance = new electionguard::CiphertextBallotSelection(
```

Comment:
in most cases a copy is required when initializing objects because the core c++ library can't assume the intent of the developer